vignettes/fims-path-maturity.Rmd

```diff
@@ -0,0 +1,614 @@
+```{Rcpp, class.source = "rcppchunk", eval = FALSE}
+//file: rcpp_maturity.hpp
+
+/*
+ the maturity_models pointer from Information that matches the 
+ id of the fims::LogisticMaturity class is equal to the pointer 
+ to fims::LogisticMaturity 
+ */
+d0->maturity_models[lm0->id] = lm0;
+```
+
+
+The `add_to_fims_tmb()` function repeats code four times to track the estimated value of parameters along with their first, second, and third derivatives. 
+
+
+```{r image-path-maturity-2, echo = FALSE, message=FALSE, out.width = '100%'}
+knitr::include_graphics("figures/fims-path-maturity-2.png")
+```
+
+
+## Inside the `fims::LogisticMaturity` class
+
+The `LogisticMaturity` class in the `fims` namespace defined in [logistic.hpp](https://github.com/NOAA-FIMS/FIMS/blob/main/inst/include/population_dynamics/maturity/functors/logistic.hpp) under [inst/include/population_dynamics/maturity/functors](https://github.com/NOAA-FIMS/FIMS/tree/main/inst/include/population_dynamics/maturity/functors) has an `evaluate` method (i.e., function) that takes an input, *x* and returns the output from a logistic function (defined in [fims_math.hpp](https://github.com/NOAA-FIMS/FIMS/blob/main/inst/include/common/fims_math.hpp) under [inst/include/common](https://github.com/NOAA-FIMS/FIMS/blob/main/inst/include/common)) using the class member `median` and `slope` values.  
+
+```{Rcpp,  class.source = "fimschunk", eval = FALSE}
+namespace fims {
+
+/**
+ *  @brief LogisticMaturity class that returns the logistic function value
+ * from fims_math.
+ */
+template <typename T>
+struct LogisticMaturity : public MaturityBase<T> {
+  T median; /*!< 50% quantile of the value of the quantity of interest (x); e.g.
+               age at which 50% of the fish are mature */
+  T slope;  /*!<scalar multiplier of difference between quantity of interest
+               value (x) and median */
+
+  LogisticMaturity() : MaturityBase<T>() {}
+
+  /**
+   * @brief Method of the logistic maturity class that implements the
+   * logistic function from FIMS math.
+   *
+   * @param x  The independent variable in the logistic function (e.g., age or
+   * size at maturity).
+   */
+  virtual const T evaluate(const T& x) {
+    return fims::logistic<T>(median, slope, x);
+  }
+};
+
+}  // namespace fims
+```
+
+```{r image-path-maturity-3, echo = FALSE, message=FALSE, out.width = '100%'}
+knitr::include_graphics("figures/fims-path-maturity-3.png")
+```
+
+## Calling maturity from population
+
+The `Population` class defined in [population.hpp](https://github.com/NOAA-FIMS/FIMS/blob/main/inst/include/population_dynamics/population/population.hpp) under [inst/include/population_dynamics/population](https://github.com/NOAA-FIMS/FIMS/blob/main/inst/include/population_dynamics/population) is where all the biological calculations happen, producing expected values used in likelihood equations and derived values important for management (e.g. SSB). 
```

Comment:
fix: change "SSB" to "spawning biomass" because SSB is not defined in this vignette.